### PR TITLE
GCP IAM used iam api instead of admin api and set IAM Resource isDeleted

### PR DESCRIPTION
**Pull request by ganeshjagdale9545**

### CODE REVIEW

File: cartography/intel/gcp/iam.py [hunk 1]
```diff
@@ -638,64 +638,163 @@ def load_bindings(neo4j_session: neo4j.Session, bindings: List[Dict], project_id
...
+        UserEmail=user['email'],
+        UserName=user['name'],
+        isDeleted=user.get('is_deleted', False),
         gcp_update_tag=gcp_update_tag,
+        project_id=project_id,
     )
 
 
 @timeit
 def attach_role_to_service_account(
```
Comment (mpurusottamc) [inline cartography/intel/gcp/iam.py:787]: do we need this new parameter `is_deleted`? we should follow the same pattern as attach_role_to_user, where we read the `is_deleted` from the user object.  In this case, we will read from service account.

**Review by mpurusottamc — COMMENTED** [review]
one minor comment. Rest of the code looks good.

File: cartography/intel/gcp/iam.py [hunk 2]
```diff
@@ -638,116 +638,256 @@ def load_bindings(neo4j_session: neo4j.Session, bindings: List[Dict], project_id
...
+                }
                 attach_role_to_user(
-                    neo4j_session, role_id, f"projects/{project_id}/users/{member[len('user:'):]}",
+                    neo4j_session, role_id, user,
                     project_id, gcp_update_tag,
                 )
 
             elif member.startswith('serviceAccount:'):
+                serviceAccount = {
+                    'id': f"projects/{project_id}/serviceAccounts/{member[len('serviceAccount:'):]}"
```
Comment (mpurusottamc) [inline cartography/intel/gcp/iam.py:655]: do we also need to populate other fields as well like name, email, etc? I see we do that for user, group, domain, etc.

File: cartography/intel/gcp/iam.py [hunk 3]
```diff
@@ -638,116 +638,256 @@ def load_bindings(neo4j_session: neo4j.Session, bindings: List[Dict], project_id
...
+        isDeleted=user.get('is_deleted', False),
         gcp_update_tag=gcp_update_tag,
+        project_id=project_id,
     )
 
 
 @timeit
 def attach_role_to_service_account(
-    neo4j_session: neo4j.Session, role_id: str,
+    neo4j_session: neo4j.Session, serviceAccount: Dict,
```
Comment (mpurusottamc) [inline cartography/intel/gcp/iam.py:791]: did we delete the role_id field by mistake? And also, i see serviceAccount Dictionary and service_account_id string parameter.

File: cartography/intel/gcp/iam.py [hunk 4]
```diff
@@ -638,116 +638,256 @@ def load_bindings(neo4j_session: neo4j.Session, bindings: List[Dict], project_id
...
-    MATCH (sa:GCPPrincipal{id:{saId}})
     MERGE (sa)-[r:ASSUME_ROLE]->(role)
     ON CREATE SET r.firstseen = timestamp()
     SET r.lastupdated = {gcp_update_tag}
     """
 
     neo4j_session.run(
         ingest_script,
-        RoleId=role_id,
+        RoleId=serviceAccount['id'],
```
Comment (mpurusottamc) [inline cartography/intel/gcp/iam.py:807]: can you double check this mapping?

**Review by mpurusottamc — CHANGES_REQUESTED** [review]
minor comments. need to make few changes.

File: cartography/intel/gcp/iam.py (hunk 2 above)
Comment (ganeshjagdale9545) [reply cartography/intel/gcp/iam.py:655]: To load Service Accounts we already used the iam api so in `attach_role_to_service_account` function we need to only match this Service Account and attached to the role

**Comment by mpurusottamc** [comment]
Thanks for clarifying. Merging now.

**Pull request merged**